x-pack/legacy/plugins/uptime/common/runtime_types/ping/ping.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+
+export const HttpResponseBodyType = t.partial({
+  bytes: t.number,
+  content: t.string,
+  content_bytes: t.number,
+  hash: t.string,
+});
+
+export type HttpResponseBody = t.TypeOf<typeof HttpResponseBodyType>;
+
+export const TlsType = t.partial({
+  certificate_not_valid_after: t.string,
+  certificate_not_valid_before: t.string,
+});
+
+export type Tls = t.TypeOf<typeof TlsType>;
+
+export const PingType = t.intersection([
+  t.type({
+    '@timestamp': t.string,
+    monitor: t.intersection([
+      t.type({
+        duration: t.type({
+          us: t.number,
+        }),
+        id: t.string,
+        status: t.string,
+        type: t.string,
+      }),
+      t.partial({
+        check_group: t.string,
+        ip: t.string,
+        name: t.string,
+        timespan: t.partial({
+          gte: t.string,
+          lte: t.string,
+        }),
+      }),
+    ]),
+  }),
+  t.partial({
+    agent: t.intersection([
+      t.type({
+        ephemeral_id: t.string,
+        hostname: t.string,
+        id: t.string,
+        type: t.string,
+        version: t.string,
+      }),
+      t.partial({
+        name: t.string,
+      }),
+    ]),
+    container: t.partial({
+      id: t.string,
+      image: t.partial({
+        name: t.string,
+        tag: t.string,
+      }),
+      name: t.string,
+      runtime: t.string,
+    }),
+    ecs: t.partial({
+      version: t.string,
+    }),
+    error: t.intersection([
+      t.partial({
+        code: t.string,
+        id: t.string,
+        stack_trace: t.string,
+        type: t.string,
+      }),
+      t.type({
+        // this is _always_ on the error field
+        message: t.string,
+      }),
+    ]),
+    http: t.partial({
+      request: t.partial({
+        body: t.partial({
+          bytes: t.number,
+          content: t.partial({
+            text: t.string,
+          }),
+        }),
+        bytes: t.number,
+        method: t.string,
+        referrer: t.string,
+      }),
+      response: t.partial({
+        body: HttpResponseBodyType,
+        bytes: t.number,
+        redirects: t.string,
+        status_code: t.number,
+      }),
+      version: t.string,
+    }),
+    icmp: t.partial({
+      requests: t.number,
+      rtt: t.partial({
+        us: t.number,
+      }),
+    }),
+    kubernetes: t.partial({
+      pod: t.partial({
+        name: t.string,
+        uid: t.string,
+      }),
+    }),
```

Comment:
Do we need these types? or we using this data?

Reply:
We do use `kubernetes` and `docker` for the integration links to APM/Logs/Metrics. I met with @andrewvc and we were looking to delete them as well but remembered that we do use them rarely.